api/src/main/java/org/eclipse/microprofile/rest/client/spi/RestClientBuilderResolver.java

```diff
@@ -95,19 +95,21 @@ private static RestClientBuilderResolver loadSpi(ClassLoader cl) {
             return null;
         }
 
-        RestClientBuilderResolver instance = null;
-
-        ServiceLoader<RestClientBuilderResolver> sl = ServiceLoader.load(RestClientBuilderResolver.class, cl);
-        for (RestClientBuilderResolver spi : sl) {
-            if (instance != null) {
-                throw new IllegalStateException("Multiple RestClientBuilderResolver implementations found: "
-                        + spi.getClass().getName() + " and "
-                        + instance.getClass().getName());
+        final PrivilegedAction<RestClientBuilderResolver> action = () -> {
+            ServiceLoader<RestClientBuilderResolver> sl = ServiceLoader.load(RestClientBuilderResolver.class, cl);
+            RestClientBuilderResolver instance = null;
+            for (RestClientBuilderResolver spi : sl) {
+                if (instance != null) {
+                    throw new IllegalStateException("Multiple RestClientBuilderResolver implementations found: "
+                            + spi.getClass().getName() + " and "
+                            + instance.getClass().getName());
+                }
+                instance = spi;
             }
-            instance = spi;
-        }
+            return instance;
+        };
 
-        return instance;
+        return System.getSecurityManager() == null ? action.run() : AccessController.doPrivileged(action);
```

Comment:
It is not wise to introduce a new usage of SecurityManager which is Deprecated for removal in JDK 18. This limits the usability in an LTS JDK21+.

Reply:
There are users that require the security manager and this already has references to the security manager. It was just missed in this one spot.

Reply:
We can merge this one and then clean up all SecurityManager usage when it is removed from JDK in due course.